src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -811,7 +811,7 @@ abstract class UnCurry extends InfoTransform
       }
 
       // check if the method with that name and those arguments already exists in the template
-      enteringUncurry(currentClass.info.member(forwSym.name).alternatives.find(s => s != forwSym && s.tpe.matches(forwSym.tpe))) match {
+      enteringUncurry(currentClass.info.member(forwSym.name).filter(s => s != forwSym && s.tpe.matches(forwSym.tpe)).alternatives.headOption) match {
```

Comment:
I don't understand how this fixes the bug - why are the two versions not the same?